src/modules/settings/containers/professional-settings-location/professional-settings-location.tsx

```diff
@@ -1,16 +1,90 @@
-import { type FC } from 'react';
-
-import { ProfileSettingsTabContentLayout } from '@/modules/user/components/profile-settings-tab-content-layout';
+import { type FC, useCallback, useRef } from 'react';
+// containers
 import { UserLocationSelectForm } from '@/modules/location/containers/user-location-select-form';
+// components
+import { ProfileSettingsTabContentLayout } from '@/modules/user/components/profile-settings-tab-content-layout';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+// types
+import type { Address } from '@/modules/location/types/address.types';
 
 export const ProfessionalSettingsLocation: FC = () => {
+  // queries
+  const { data: me } = trpc.user.me.useQuery({ expand: ['professional'] });
+  const { data: location, ...locationQuery } =
+    trpc.location.getByProfessionalId.useQuery(
+      {
+        id: me?.professional?.id ?? '',
+      },
+      {
+        enabled: Boolean(me?.professional?.id),
+        retry: (retryCount, error) => {
+          if (error.data?.code === 'NOT_FOUND') return false;
+
+          return retryCount <= 2;
+        },
+      }
+    );
+  // mutations
+  const locationCreate = trpc.location.create.useMutation();
+  const locationUpdate = trpc.location.update.useMutation();
+  const locationDelete = trpc.location.delete.useMutation();
+
+  // refs
+  const locationSelectFormRef = useRef<null | {
+    getAddress: () => Address | null;
+  }>(null);
+  const isSaveLoading =
+    locationCreate.isLoading ||
+    locationUpdate.isLoading ||
+    locationDelete.isLoading;
+
+  const handleSave = useCallback(async () => {
+    if (!locationSelectFormRef.current) return;
+
+    const address = locationSelectFormRef.current.getAddress();
+
+    if (!location && address) {
```

Comment:
It's still not clear what's the difference between address and location if looking at your comment. I checked the code and it seems like address it's just a new location that user select in the user location selector. In my opinion it would be better to change address to `newLocation` so we will understand that it's something new that user selected compared to old location that we had.